x/deployment/query/querier.go

```diff
@@ -24,31 +29,46 @@ func NewQuerier(keeper keeper.Keeper) sdk.Querier {
 	}
 }
 
-func queryDeployments(ctx sdk.Context, path []string, _ abci.RequestQuery, keeper keeper.Keeper) ([]byte, error) {
+func queryDeployments(ctx sdk.Context, path []string, _ abci.RequestQuery, k keeper.Keeper) ([]byte, error) {
 	// isValidState denotes whether given state flag is valid or not
 	filters, isValidState, err := parseDepFiltersPath(path)
 	if err != nil {
 		return nil, sdkerrors.Wrap(types.ErrInternal, err.Error())
 	}
 
 	var values Deployments
-	keeper.WithDeployments(ctx, func(deployment types.Deployment) bool {
-		if filters.Accept(deployment, isValidState) {
+
+	if !filters.After {
+		k.WithDeployments(ctx, func(deployment types.Deployment) bool {
+			if filters.Accept(deployment, isValidState) {
+				value := Deployment{
+					Deployment: deployment,
+					Groups:     k.GetGroups(ctx, deployment.ID()),
+				}
+				values = append(values, value)
+			}
```

Comment:
I think we want to use filters in both cases.  Ideally we wouldn't have to do this same chunk of code twice.